### ISSUE TITLE
Run "ceph-salt apply" sanity/pre-flight checks in "ceph-salt status" as well?

### ISSUE THREAD

**Issue by smithfarm** [ceph/ceph-salt#379]
Currently `ceph-salt status` checks that the configuration entered by the user is internally coherent, and `ceph-salt apply` checks the actual nodes for problematic states that are known to make `ceph-salt apply` fail.

I can imagine that users might experience consternation if they (1) run `ceph-salt status` and see OK, (2) run `ceph-salt apply` and it fails in the sanity/pre-flight checks.

Would it be feasible for `ceph-salt status` to also run the `ceph-salt apply` sanity/pre-flight checks?

**Comment by ricardoasmarques** [comment]
Agree. When I see `ceph-salt status` returning `OK`, I'm expecting that `ceph-salt apply` will work.

**Comment by smithfarm** [comment]
We agreed that:

1. `ceph-salt apply` sanity checks should be expanded to include the checks currently being done by `ceph-salt status`
2. `ceph-salt status` should be re-implemented as a "dry run" of `ceph-salt apply` (do everything up to, but not including applying the Salt Formula)
